adapter/src/hermes/adapter/stdio/stdio.cc

```diff
@@ -0,0 +1,992 @@
+        auto file_read_size =
+            perform_file_read(filename.c_str(), item.first.offset_, ptr,
+                              total_read_size, item.second.size_);
+        read_size += file_read_size;
+      }
+      if (contains_blob && fs::exists(filename) &&
+          fs::file_size(filename) >= item.first.offset_ + item.first.size_) {
+        LOG(INFO) << "Blob does not have data and need to read from original "
+                     "filename: "
+                  << filename << " offset:" << item.first.offset_ + read_size
+                  << " size:" << item.second.size_ - read_size << "."
+                  << std::endl;
+        auto new_read_size = perform_file_read(
+            filename.c_str(), item.first.offset_, ptr,
+            total_read_size + read_size, item.second.size_ - read_size);
+        read_size += new_read_size;
+      }
+    } else if (fs::exists(filename) &&
+               fs::file_size(filename) >=
+                   item.first.offset_ + item.first.size_) {
+      LOG(INFO)
+          << "Blob does not exists and need to read from original filename: "
+          << filename << " offset:" << item.first.offset_
+          << " size:" << item.first.size_ << "." << std::endl;
+      read_size = perform_file_read(filename.c_str(), item.first.offset_, ptr,
+                                    total_read_size, item.first.size_);
+    }
+    if (read_size > 0) {
+      total_read_size += read_size;
+    }
+  }
+  existing.first.st_ptr += total_read_size;
+  struct timespec ts;
+  timespec_get(&ts, TIME_UTC);
+  existing.first.st_atim = ts;
+  existing.first.st_ctim = ts;
+  mdm->Update(fp, existing.first);
+  ret = total_read_size;
+  return ret;
+}
+
+FILE *HERMES_DECL(fopen)(const char *path, const char *mode) {
+  FILE *ret;
+  std::string path_str(path);
+  if (hermes::adapter::IsTracked(path)) {
+    LOG(INFO) << "Intercept fopen for filename: " << path
+              << " and mode: " << mode << " is tracked." << std::endl;
+    ret = open_internal(path, mode);
+  } else {
+    MAP_OR_FAIL(fopen);
+    ret = real_fopen_(path, mode);
+  }
+  return (ret);
+}
+
+FILE *HERMES_DECL(fopen64)(const char *path, const char *mode) {
+  FILE *ret;
+  std::string path_str(path);
+  if (hermes::adapter::IsTracked(path)) {
+    LOG(INFO) << "Intercept fopen64 for filename: " << path
+              << " and mode: " << mode << " is tracked." << std::endl;
+    ret = open_internal(path, mode);
+  } else {
+    MAP_OR_FAIL(fopen64);
+    ret = real_fopen_(path, mode);
+  }
+  return (ret);
+}
+
+FILE *HERMES_DECL(fdopen)(int fd, const char *mode) {
+  FILE *ret;
+  MAP_OR_FAIL(fdopen);
+  ret = real_fdopen_(fd, mode);
+  if (ret && hermes::adapter::IsTracked(ret)) {
+    LOG(INFO) << "Intercept fdopen for file descriptor: " << fd
+              << " and mode: " << mode << " tracked." << std::endl;
+    int MAXSIZE = 0xFFF;
+    char proclnk[0xFFF];
+    char filename[0xFFF];
```

Comment:
Use `MAXSIZE` instead of repeating literals.

Reply:
Fixed